auth-api/src/test/java/io/stargate/auth/api/AuthResourceTest.java

```diff
@@ -0,0 +1,244 @@
+class AuthResourceTest {
+
+    private static final AuthenticationService authService = mock(AuthenticationService.class);
+
+    private static final ResourceExtension resourceWithUsernameTokenDisabled = ResourceExtension.builder()
+            .addResource(new AuthResource(authService, false))
+            .build();
+
+    private static final ResourceExtension resourceWithUsernameTokenEnabled = ResourceExtension.builder()
+            .addResource(new AuthResource(authService, true))
+            .build();
+
+    @AfterEach
+    void tearDown() {
+        reset(authService);
+    }
+
+    @Test
+    void createTokenFromSecretSuccess() throws UnauthorizedException {
+        Secret secret = new Secret("key", "secret");
+        when(authService.createToken("key", "secret")).thenReturn("token");
+
+        AuthTokenResponse token = resourceWithUsernameTokenDisabled.target("/v1/auth/token/generate")
+                .request()
+                .post(Entity.entity(secret, MediaType.APPLICATION_JSON), AuthTokenResponse.class);
+
+        assertThat(token.getAuthToken()).isEqualTo("token");
+    }
+
+    @Test
+    void createTokenFromSecretUnauthorized() throws UnauthorizedException {
+        Secret secret = new Secret("key", "secret");
+        when(authService.createToken("key", "secret")).thenThrow(UnauthorizedException.class);
+
+        Response response = resourceWithUsernameTokenDisabled.target("/v1/auth/token/generate")
+                .request()
+                .post(Entity.entity(secret, MediaType.APPLICATION_JSON));
+
+        assertThat(response.getStatusInfo().getStatusCode()).isEqualTo(401);
+    }
+
+    @Test
+    void createTokenFromSecretInternalServerError() throws UnauthorizedException {
+        Secret secret = new Secret("key", "secret");
+        when(authService.createToken("key", "secret")).thenThrow(RuntimeException.class);
+
+        Response response = resourceWithUsernameTokenDisabled.target("/v1/auth/token/generate")
+                .request()
+                .post(Entity.entity(secret, MediaType.APPLICATION_JSON));
+
+        assertThat(response.getStatusInfo().getStatusCode()).isEqualTo(500);
+    }
+
+    @Test
+    void createTokenFromSecretNoPayload() {
+        Response response = resourceWithUsernameTokenDisabled.target("/v1/auth/token/generate")
+                .request()
+                .post(null);
+
+        assertThat(response.getStatusInfo().getStatusCode()).isEqualTo(400);
+    }
+
+    @Test
+    void createTokenFromSecretEmptyKey() {
+        Secret secret = new Secret("", "secret");
+
+        Response response = resourceWithUsernameTokenDisabled.target("/v1/auth/token/generate")
+                .request()
+                .post(Entity.entity(secret, MediaType.APPLICATION_JSON));
+
+        assertThat(response.getStatusInfo().getStatusCode()).isEqualTo(400);
+    }
+
+    @Test
+    void createTokenFromSecretEmptySecret() {
+        Secret secret = new Secret("key", "");
```

Comment:
Since we are adding extensive tests for error conditions, how about testing malformed input JSON (including _unset_ required fields)?

Reply:
I'd be fine with testing a single 400 case for now. Maybe we can add it later if we feel that it is needed?

Reply:
I believe testing malformed input is critical for security API. Since this PR adds _some_ tests for credential validation, it is really that much work to add a malformed input test? :)

Reply:
Given the fact that there were 0 tests, I think the PR still adds value right now but I see your point. Of course it is not that much work. I will see if I can find some time on the week end. :)

Reply:
@dimas-b what do you mean by malformed json? Dropwizard is going to reject the request if it can't be serialized to the correct object.

> Given the fact that there were 0 tests

0 unit tests to clarify :)

Reply:
Right, but actual clients are not restricted by dropwizard and could potentially use malformed JSON in the POST request body. I understand it's a bit of extra work, but since we opened this can of worms by other tests... again, if it proves too much work, I'd be fine with deferring adding those tests (with a separate GH issue perhaps).

Reply:
Also, what happens if the JSON is well formed, but represents something other than a proper `Secret` object?.. e.g. has extra data in it.

Reply:
> @dimas-b what do you mean by malformed json? Dropwizard is going to reject the request if it can't be serialized to the correct object.
> 
> > Given the fact that there were 0 tests
> 
> 0 unit tests to clarify :)

Ah I didn't find those. Can you point me to them? I'm probably blind 🙈

Reply:
https://github.com/stargate/stargate/blob/master/testing/src/test/java/io/stargate/it/http/AuthApiTest.java

Reply:
Whoops. A lot of duplicate test cases it seems :D I was misguided by the oss labeled issue https://github.com/stargate/stargate/issues/83 which had no further description. Should we close this PR then?